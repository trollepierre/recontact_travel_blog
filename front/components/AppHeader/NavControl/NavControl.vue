<template>
  <ul class="navigation">
    <li
      v-if="showNavBarButton"
      class="link">
      <app-button
        class="subscribe"
        :text="$t('subscribe')"
        @click="displaySubscribeModal"/>
    </li>
    <li
      v-if="showNavBarButton"
      class="link">
      <app-button
        class="suggestion"
        :text="$t('suggestion')"
        @click="displayFeedbackModal"/>
    </li>
    <li
      v-if="showNavBarButton"
      class="link tdm">
      <a
        :title="$t('tdm')"
        class="tdm"
        href="http://worldtour.recontact.me">
        <img
          :alt="$t('logo')"
          class="tdm__image"
          src="../../../static/tdm.jpg">
      </a>
    </li>
    <li>
      <app-button :hide="true"/>
    </li>
    <li class="link new-link">
      <new-mode-toggle/>
    </li>
    <li class="link dark-link">
      <dark-mode-toggle/>
    </li>
    <li class="link other-language">
      <app-button
        class="other-language"
        :text="$t('otherLanguage')"
        @click="switchLanguage"/>
    </li>
  </ul>
</template>
<script>
  import AppButton from '@/components/AppButton/AppButton'
  import DarkModeToggle from '@/components/AppHeader/NavControl/DarkModeToggle/DarkModeToggle'
  import NewModeToggle from '@/components/AppHeader/NavControl/NewModeToggle/NewModeToggle'

  export default {
    name: 'NavControl',
    components: { AppButton, DarkModeToggle, NewModeToggle },
    data: () => ({
      showNavBarButton: false,
    }),
    methods: {
      switchLanguage() {
        window.location.href = this.$t('otherUrl') + window.location.pathname
      },
      displaySubscribeModal() {
      // this.$modal.show('subscribe-modal')
      },
      displayFeedbackModal() {
      // this.$modal.show('feedback-modal')
      },
    },
    i18n: {
      silentTranslationWarn: true,
      messages: {
        fr: {
          subscribe: 'Sâ€™abonner',
          suggestion: 'Laisser un message',
          tdm: 'Retrouver lâ€™ancien site du tour du monde de Pierre et BenoÃ®t',
          logo: 'Logo',
          otherLanguage: 'ðŸ‡¬ðŸ‡§',
          otherUrl: 'https://www.recontact.me',
        },
        en: {
          subscribe: 'Subscribe',
          suggestion: 'Leave a message',
          tdm: 'Go to see the former website of the world trip of Pierre and BenoÃ®t',
          logo: 'Logo',
          otherLanguage: 'ðŸ‡«ðŸ‡·',
          otherUrl: 'https://fr.recontact.me',
        },
      },
    },
  }
</script>
<style lang="scss" scoped>

.navigation {
  list-style: none;
  margin: 0;
  display: inline-flex;
  padding-left: 0;
  height: 100%;
}

.suggestion,
.tdm {
  display: none;
}

.tdm {
  display: inline-flex;
  padding: 10px;
}

.tdm__image {
  color: $tdm-color;
}

.button {
  background-color: $header-bg;
}

.button:hover {
  background-color: $button-hover;
}

.button.other-language { // side effect
  font-size: 24px;
  margin-top: 0;
  padding: 8px 6px 0 6px;
}

.button.other-language:hover {
  background-color: $button-hover;
}

.other-language, .dark-link, .new-link {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 5px;
}

@media only screen and (max-width: 340px) {
  .button.other-language {
    font-size: 10px;
  }
}

@media only screen and (min-width: 640px) {
  .suggestion,
  .tdm {
    display: inline-block;
  }

  .link {
    margin-left: 25px;
  }
}

@media only screen and (min-width: 640px) {
  .button.other-language {
    padding: 5px 8px 0;
  }
}
</style>
