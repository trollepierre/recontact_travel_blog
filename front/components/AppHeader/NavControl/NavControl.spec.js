import VueI18n from 'vue-i18n'
import NavControl from './NavControl.vue'

describe('Component | NavControl.vue', () => {
  let localVue
  let wrapper

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VueI18n)
  })

  describe('template', () => {
    it('should match snapshot', () => {
      wrapper = shallowMount(NavControl, { localVue })

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('methods', () => {
    describe('switchLanguage', () => {
      it('should reload page', () => {
        // Given
        console.error = jest.fn()
        wrapper = shallowMount(NavControl, { localVue })

        // When
        wrapper.vm.switchLanguage()

        // Then
        expect(window.location.href).toEqual('http://localhost/')
      })
    })

    xdescribe('#displayFeedbackModal', () => {
      it('should display the feedback-modal', () => {
        wrapper.vm.$modal.show = jest.fn()

        wrapper.vm.displayFeedbackModal()

        expect(wrapper.vm.$modal.show).toHaveBeenCalledWith('feedback-modal')
      })
    })

    xdescribe('#displaySubscribeModal', () => {
      it('should display the subscribe-modal', () => {
        wrapper.vm.$modal.show = jest.fn()

        wrapper.vm.displaySubscribeModal()

        expect(wrapper.vm.$modal.show).toHaveBeenCalledWith('subscribe-modal')
      })
    })
  })

  describe('locales', () => {
    const languages = Object.keys(NavControl.i18n.messages)

    it('contains 2 languages', () => {
      expect(languages).toHaveLength(2)
      expect(languages).toEqual(['fr', 'en'])
    })

    describe('each language', () => {
      it('contains fr locales', () => {
        const locales = Object.keys(NavControl.i18n.messages.fr)
        expect(locales).toMatchInlineSnapshot(`
          Array [
            "subscribe",
            "suggestion",
            "tdm",
            "logo",
            "otherLanguage",
            "otherUrl",
          ]
        `)
      })

      it('contains en locales', () => {
        const locales = Object.keys(NavControl.i18n.messages.en)
        expect(locales).toMatchInlineSnapshot(`
          Array [
            "subscribe",
            "suggestion",
            "tdm",
            "logo",
            "otherLanguage",
            "otherUrl",
          ]
        `)
      })
    })
  })
})
