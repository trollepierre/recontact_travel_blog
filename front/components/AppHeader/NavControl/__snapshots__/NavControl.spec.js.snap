// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | NavControl.vue template should match snapshot 1`] = `
<ul class="navigation">
  <!---->
  <!---->
  <!---->
  <li>
    <app-button-stub text="" tag="button" hide="true"></app-button-stub>
  </li>
  <li class="link new-link">
    <new-mode-toggle-stub></new-mode-toggle-stub>
  </li>
  <li class="link dark-link">
    <dark-mode-toggle-stub></dark-mode-toggle-stub>
  </li>
  <li class="link other-language">
    <app-button-stub text="ðŸ‡«ðŸ‡·" tag="button" class="other-language"></app-button-stub>
  </li>
</ul>
`;
